src/large_pages/node_large_page.cc

```diff
@@ -257,22 +257,34 @@ struct text_region FindNodeTextRegion() {
 }
 
 #if defined(__linux__)
-bool IsTransparentHugePagesEnabled() {
-  std::ifstream ifs;
+std::istream* GetTransparentHugePagesConfiguration() {
+  std::ifstream* ifs = new std::ifstream();
 
-  ifs.open("/sys/kernel/mm/transparent_hugepage/enabled");
-  if (!ifs) {
+  ifs->open("/sys/kernel/mm/transparent_hugepage/enabled");
+  if (!*ifs) {
     PrintWarning("could not open /sys/kernel/mm/transparent_hugepage/enabled");
-    return false;
+    return nullptr;
   }
 
-  std::string always, madvise;
-  if (ifs.is_open()) {
-    while (ifs >> always >> madvise) {}
+  return ifs;
+}
+
+bool IsTransparentHugePagesConfigured(std::istream* configuration) {
+  std::string token;
+  while (*configuration >> token) {
+    if (token == "[always]" || token == "[madvise]") return true;
   }
-  ifs.close();
+  return false;
+}
 
-  return always == "[always]" || madvise == "[madvise]";
+bool IsTransparentHugePagesEnabled() {
+  std::istream* configuration = GetTransparentHugePagesConfiguration();
+  if (configuration == nullptr) {
+    return false;
+  }
```

Comment:
TODO: rewrite this to place the `if` and `return` within a single-line; that pattern exists within the codebase already and would save a couple of lines.  Not pushing this as a commit immediately since it'd spend unnecessary continuous integration resources.